### PR TITLE
admin routes for company approval + map api

### PR DESCRIPTION
**Pull request by hamoudi-hamoudi**
## 📝 Description

Screenshot 2023-10-11 204740
Screenshot 2023-10-11 204908
Screenshot 2023-10-11 204704

### CODE REVIEW

File: frontend/src/components/adminDetails/companyCard/CompanyCard.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  const [isClicked, setIsClicked] = useState<boolean>(false);
+  const deleteCompany = useAdminStore((state) => state.deleteCompany);
+  const addcompany = useAdminStore((state) => state.addcompany);
+  const [isShowAbout, setIsShowAbout] = useState(false);
+
+  const handleDenyClick = async (id: number) => {
+    setIsClicked(true);
+    setTimeout(() => {
+      deleteCompany(id);
+    }, 300);
```
Comment (BrantalikP) [inline frontend/src/components/adminDetails/companyCard/CompanyCard.tsx:19]: Why do we need the timeout here?

File: frontend/src/map/Map.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      }
+    }
+
+    fetchCoordinates();
+  }, []);
+  useEffect(() => {
+    if (coordinates.lat !== 0 && coordinates.lng !== 0) {
+      setIsLoading(false);
+    }
+  }, [coordinates]);
```
Comment (BrantalikP) [inline frontend/src/map/Map.tsx]: there is no need for the useEffect and useState hook in this case I think, 

simply create a new variable 
`const isLoading = coordinates.lat === 0 || coordinates.lng === 0`

That should work as well :)

**Review by BrantalikP — CHANGES_REQUESTED** [review]
Nicely done 💪 Just please check out the comments and remove all unused code/unnecessary logs reported by the **Lint Bot**. Also could you share the link for the **company to approve** designs? I cannot find it :/

File: frontend/src/components/adminDetails/companyCard/CompanyCard.tsx (hunk 1 above)
Comment (hamoudi-hamoudi) [reply frontend/src/components/adminDetails/companyCard/CompanyCard.tsx:19]: Hi , If I don't use it the fade away style will not work :pleading_face: 
I tried with many approach this the only way its working

File: frontend/src/map/Map.tsx (hunk 2 above)
Comment (hamoudi-hamoudi) [reply frontend/src/map/Map.tsx]: much better solution, Thank you :grin: . on it

**Comment by hamoudi-hamoudi** [comment]
> Nicely done 💪 Just please check out the comments and remove all unused code/unnecessary logs reported by the **Lint Bot**. Also could you share the link for the **company to approve** designs? I cannot find it :/

Screenshot 2023-10-12 204216
Screenshot 2023-10-12 204240
Screenshot 2023-10-12 204257

**Comment by hamoudi-hamoudi** [comment]
https://github.com/HackYourFuture-CPH/class-25-ironside/assets/33804102/0aa68ba9-80e3-487f-8e52-349495d0954b

**Comment by hamoudi-hamoudi** [comment]
https://github.com/HackYourFuture-CPH/class-25-ironside/assets/33804102/0b29e583-0a0f-4ab8-a677-a107e0f333bd

**Review by BrantalikP — APPROVED** [review]
🚀

**Pull request merged**